language-support/ts/codegen/tests/ts/build-and-lint-test/src/__tests__/test.ts

```diff
@@ -611,3 +613,38 @@ test('package API', async () => {
   const downSuc = await ledger.getPackage(buildAndLint.packageId);
   expect(downSuc.byteLength > 0).toBe(true);
 });
+
+test('reconnect on timeout, when multiplexing is enabled', async () => {
```

Comment:
not quite happy with the test as it does a wait to trigger a timeout and re-connect , can someone suggest a cleaner/better way to do this ? want to test reconnect logic in an integration test.

Reply:
I don't really have a better idea for an integration test, no. If you could mock the server, I suppose you could fake a timeout faster, but our current fake server (in react-ledger tests) is not exactly great for that IIRC.